src/Microsoft.Identity.Client/Config/ApplicationConfiguration.cs

```diff
@@ -0,0 +1,78 @@
+﻿// ------------------------------------------------------------------------------
+// 
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+// 
+// This code is licensed under the MIT License.
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Http;
+using Microsoft.Identity.Client.TelemetryCore;
+
+namespace Microsoft.Identity.Client.Config
+{
+    internal sealed class ApplicationConfiguration : IApplicationConfiguration
```

Comment:
>internal [](start = 4, length = 8)

I wonder if this (and the Config property on the Builder) should be public. I suppose during debugging it will be easier to see what values are used by the application. thoughts?

Reply:
I don't want this to be public because no one outside of the library should be taking a dependency on this actual class.  it's built by us within the builder and then passed around internally.  But public users shouldn't be constructing this.  I"m curious to hear more about how having this public would help with debugging.

Reply:
i looked into making IAppConfig public and it spreads into other types being public.  We could have a sub-version of this interface that has the values on it and do interface derivation or just keep the properties we want to read on IClientApplicationBase and forward them out of there (this is what we do today).

---
In reply to: [241492588](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/723#discussion_r241492588) [](ancestors = 241492588,241283812)